hawk/app/models/cib.rb

```diff
@@ -1014,33 +972,21 @@ def initialize(id, user, use_file = false, stonithwarning = false)
             if ignore_failure
               failed_ops[-1][:ignored] = true
               rc_code = expected
-            else
-              if operation == "stop"
-                # We have a failed stop, the resource is failed (bnc#879034)
-                state = :failed
-                # Also, the node is thus unclean if STONITH is enabled.
-                node[:state] = :unclean if @crm_config[:stonith_enabled]
-              end
+            elsif operation == "stop"
+              # We have a failed stop, the resource is failed (bnc#879034)
+              state = :failed
+              # Also, the node is thus unclean if STONITH is enabled.
+              node[:state] = :unclean if @crm_config[:stonith_enabled]
             end
           end
 
-          # TODO(should): evil magic numbers!
-          # The operation and RC code tells us the state of the resource on this node
-          # when rc=0, anything other than a stop means we're running
-          # (might be slave after a demote)
-          # TODO(must): verify this demote business
-          case rc_code
-          when 7
-            state = :stopped
-          when 8
-            state = :master
-          when 0
-            if operation == 'stop' || operation == 'migrate_to'
-              state = :stopped
-            elsif operation == 'promote'
-              state = :master
-            else
-              state = :started
+          state = CibTools.op_rc_to_state operation, rc_code, state
+
+          # check for guest nodes
+          if !op.attributes['on_node'].nil? && [:master, :started].include?(state)
```

Comment:
The condition here doesn't look sufficient to me. IIUC, this might make a normal resource that happens to have an ID that is conflicting with a node name incorrectly fall into the case as if it's a special internal `ocf:pacemaker:remote`  resource. Other than that it's `ocf:pacemaker:remote` type, the specialty of such an internal resource is the `lrm_resource` element has a `container` attribute like:

https://github.com/ClusterLabs/pacemaker/blob/master/cts/scheduler/bug-cl-5247.xml#L251

I assume a potential solution would be add an additional `&&` here which checks `lrm_resource.attributes['container']` exists.

Please check and verify my theory.

Reply:
seems to work

Reply:
Nice!  Now we can check where/how it goes with a normal resource that happens to have the ID conflicting with a node name.

Reply:
nice thx yan.  I dunno if the actual patch should be enough:
https://github.com/ClusterLabs/hawk/pull/210/files#diff-1f508dc254aa7d0561bbef5740e5d3deR959